qiskit/primitives/statevector_estimator.py

```diff
@@ -0,0 +1,128 @@
+"""
+Estimator class
+"""
+
+from __future__ import annotations
+
+from collections.abc import Iterable, Mapping
+from dataclasses import dataclass
+from typing import Optional, Union
+
+import numpy as np
+
+from qiskit.quantum_info import SparsePauliOp, Statevector
+
+from .base import BaseEstimatorV2
+from .containers import EstimatorPub, EstimatorPubLike, PrimitiveResult, PubResult
+from .primitive_job import PrimitiveJob
+from .utils import bound_circuit_to_instruction
+
+
+@dataclass
+class Options:
+    """Options for estimator."""
+
+    seed: Optional[Union[int, np.random.Generator]] = None
+
+    def update(self, options: Options | Mapping | None = None, **kwargs):
+        """Update the options."""
+        if options is not None:
+            if isinstance(options, Mapping):
+                options_dict = options
+            elif isinstance(options, Options):
+                options_dict = options.__dict__
+            else:
+                raise TypeError(f"Type {type(options)} is not options nor Mapping class")
+            for key, val in options_dict.items():
+                setattr(self, key, val)
+
+        for key, val in kwargs.items():
+            setattr(self, key, val)
+
+
+class Estimator(BaseEstimatorV2):
+    """
+    Simple implementation of :class:`BaseEstimatorV2` with Statevector.
+    """
+
+    def __init__(self, options: Options | dict | None = None):
+        if options is None:
+            options = Options()
+        elif not isinstance(options, Options):
+            options = Options(**options)
+        self.options = options
+
+    def run(
+        self, pubs: Iterable[EstimatorPubLike], precision: float | None = None
+    ) -> PrimitiveJob[PrimitiveResult[PubResult]]:
+        job: PrimitiveJob[PrimitiveResult[PubResult]] = PrimitiveJob(self._run, pubs, precision)
+        job._submit()
+        return job
+
+    def _run(
+        self, pubs: Iterable[EstimatorPub], precision: float | None
+    ) -> PrimitiveResult[PubResult]:
+        coerced_pubs = [EstimatorPub.coerce(pub) for pub in pubs]
+
+        for pub in coerced_pubs:
+            pub.validate()
+
+        rng = _get_rng(self.options.seed)
+        run_precision = precision
+
+        results = []
+        for pub in coerced_pubs:
+            if pub.precision is not None:
+                precision = pub.precision
+            else:
+                precision = run_precision
+
+            circuit = pub.circuit
+            observables = pub.observables
+            parameter_values = pub.parameter_values
+            bound_circuits = parameter_values.bind_all(circuit)
+
+            bc_circuits, bc_obs = np.broadcast_arrays(bound_circuits, observables)
+            evs = np.zeros_like(bc_circuits, dtype=np.float64)
+            stds = np.zeros_like(bc_circuits, dtype=np.float64)
+            for index in np.ndindex(*bc_circuits.shape):
+                bound_circuit = bc_circuits[index]
```

Comment:
I think this will work and will prevent the need to have all bound circuits in memory at once.

```suggestion
                bound_circuit = parameter_values.bind(circuit, loc=index)
```